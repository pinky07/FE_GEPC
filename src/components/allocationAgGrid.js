import React from 'react';
import { connect } from 'react-redux';
import {
  Row,
  Col,
} from 'reactstrap';
import {AgGridReact} from "ag-grid-react";
import allocationModel from '../model/allocationModel';
import MixDetails from './mixDetails';
import MixStatistics from './mixStatistics';

let columnDefs = [
  {
    field: 'accountgroupname',
    headerName: 'Node Name',
    width: 350,
  },
  {
    field: 'assetCat',
    headerName: 'Asset Category',
    width: 150,
  },
  {
    field: 'newGrouping',
    headerName: 'New Grouping',
    width: 150,
  },
  {
    field: 'aamb',
    headerName: 'AAMB',
    width: 150,
  },
  {
    field: 'nd',
    headerName: 'Node',
    width: 150,
  },
  {
    field: 'alias',
    headerName: 'Asset Class Alias',
    width: 150,
  },
  {
    field: 'policy',
    headerName: 'Policy',
    width: 150,
  },
  {
    field: 'value',
    headerName: 'Value',
    width: 150,
  },
  {
    field: 'mixA',
    headerName: 'Mix A',
    width: 150,
    editable: true,
  },
  {
    field: 'mixB',
    headerName: 'Mix B',
    width: 150,
    editable: true,
  },
  {
    field: 'mixC',
    headerName: 'Mix C',
    width: 150,
    editable: true,
  }
];

class Aggrid extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      columnDefs
    }
  }
  componentWillReceiveProps (nextProps) {
    if (nextProps.gridData !== this.props.gridData) {
      allocationModel().getAllocationGrid(nextProps.gridData).then(rows => { console.log(rows)
        this.setState({
          rows,
        });
      });
    }
  }

  onGridReady(params) {
    this.gridApi = params.api;
    this.columnApi = params.columnApi;
    this.gridApi.sizeColumnsToFit();
  }

  render () {

    return (
      <div className="allocationGrid">
        <Row>
          <Col lg="9" md="9" className="gridContainer">
            <div className="aggridcontainer ag-fresh">
              <AgGridReact
                columnDefs={this.state.columnDefs}
                rowData={this.state && this.state.rows}
                onGridReady={this.onGridReady}
                headerHeight="35"
              />
            </div>
          </Col>
          <Col lg="3" md="3">
            <MixStatistics />
            <MixDetails />
          </Col>
        </Row>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    ...state.allocationGrid
  };
};

export default connect(mapStateToProps, null)(Aggrid);
