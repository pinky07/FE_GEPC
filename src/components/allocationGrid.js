import React from 'react';
import { connect } from 'react-redux';
import createReactClass from 'create-react-class';
import {
  Row,
  Col,
} from 'reactstrap';
import ReactDataGrid from 'react-data-grid';
import update from 'immutability-helper';
import RowRenderer from './rowRenderer';
import allocationModel from '../model/allocationModel';
import ExpanderFormatter from './colNameFormatter';
import MixDetails from './mixDetails';
import MixStatistics from './mixStatistics';
import BetaGroupDropdown from './betaGroupDropdown';

let columns = [
  {
    key: 'node',
    name: 'Node Name',
    width: 350,
    formatter: ExpanderFormatter
  },
  {
    key: 'assetCat',
    name: 'Asset Category',
    width: 150,
  },
  {
    key: 'newGrouping',
    name: 'New Grouping',
    width: 150,
    editable: true
  },
  {
    key: 'aamb',
    name: 'AAMB',
    width: 150,
  },
  {
    key: 'nd',
    name: 'Node',
    width: 150,
  },
  {
    key: 'alias',
    name: 'Asset Class Alias',
    width: 150,
  },
  {
    key: 'policy',
    name: 'Policy',
    width: 100,
  },
  {
    key: 'value',
    name: 'Value',
    width: 100,
  },
  {
    key: 'mixA',
    name: 'Mix A',
    width: 100,
  }
];

const AllocationGrid = createReactClass({
  getInitialState() {
    return { expanded: {}, rows: [] };
  },

  getRows(i) {
    return this.state.rows[i];
  },

  handleGridRowsUpdated({ fromRow, toRow, updated }) {
    let rows = this.state.rows.slice();

    for (let i = fromRow; i <= toRow; i++) {
      let rowToUpdate = rows[i];
      let updatedRow = update(rowToUpdate, {$merge: updated});
      rows[i] = updatedRow;
    }

    this.setState({ rows });
  },

  componentWillReceiveProps (nextProps) {
    if (nextProps.gridData !== this.props.gridData) {
      allocationModel().getAllocationGrid(nextProps.gridData).then(rows => {
        this.setState({
          rows,
        });
      });
    }
  },

  render() {
    return (
      <div className="allocationGrid">
        <Row>
          <Col lg="12" md="12" className="betagroupDropdown">
            <BetaGroupDropdown/>
          </Col>
        </Row>
        <Row>
          <Col lg="9" md="9" className="gridContainer">
            <ReactDataGrid
              enableCellSelect={true}
              columns={columns}
              rowGetter={this.getRows}
              rowsCount={this.state.rows.length}
              minHeight={500}
              rowRenderer={RowRenderer}
              onGridRowsUpdated={this.handleGridRowsUpdated}
            />
          </Col>
          <Col lg="3" md="3">
            <MixStatistics />
            <MixDetails />
          </Col>
        </Row>
      </div>
    );
  }
});

const mapStateToProps = state => {
  return {
    ...state.allocationGrid
  };
};

export default connect(mapStateToProps, null)(AllocationGrid);
