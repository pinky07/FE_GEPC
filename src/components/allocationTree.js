import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { ContextMenuProvider } from 'react-contexify';
import {
  Row,
  Col,
  Input,
} from 'reactstrap';
import SortableTree from 'react-sortable-tree';
import {
  getAllocationTree,
  saveAllocationTree,
  selectNode,
  updateTree,
} from '../actions';
import Constants from '../services/constants';
import NodeMenu from './nodeMenu';
import TreeButtonBar from './treeButtonBar';
import NodeDetails from './nodeDetails';
import BetaGroupDropdown from './betaGroupDropdown';
import TreeNodeRenderer from './treeNodeRenderer';

export class AllocationTree extends React.Component {

  componentDidMount () {
    this.props.getAllocationTree();
  }

  componentWillReceiveProps (nextProps) {
    if (nextProps.tree !== this.props.tree) {
      if (!this.props.selectedNode) {
        this.props.selectNode({ node: nextProps.tree.data[0] });
      }
    }
  }

  saveTree = () => {
    this.props.saveAllocationTree(this.props.tree);
  }

  treeButtons = rowInfo => {
    const buttons = [
      <ContextMenuProvider id="menu_id" event="onClick">
        <button
          onClick={() => this.props.selectNode(rowInfo)}
        >
          ...
        </button>
      </ContextMenuProvider>,
      <button
        className="buttonInfoNode"
        onClick={() => this.props.selectNode(rowInfo)}
      >
        ...
      </button>,
    ];

    return buttons;
  }

  render () {
    const { tree } = this.props;
    return (
      <div className="allocationTree">
        <Row className="treeHeader">
          <Col lg="3" md="3" sm="12" xs="12">
            <h5>{tree.name}</h5>
          </Col>

          <Col lg="5" md="5" sm="12" xs="12">
            <TreeButtonBar
              save={this.saveTree}
            />
          </Col>
          <Col lg="4" md="4" sm="12" xs="12">
            <div className="settings">
              <Input placeholder="Global Set" />
              <BetaGroupDropdown />
            </div>
          </Col>
        </Row>

        <Row>
          <Col lg="8" md="8" sm="12" className="treecontainer">
            <SortableTree
              treeData={tree.data}
              onChange={treeData => this.props.updateTree(treeData)}
              canDrop={ ({ nextPath }) => nextPath.length > 1}
              maxDepth={Constants.MAX_TREE_DEPTH}
              generateNodeProps={rowInfo => ({
                buttons: this.treeButtons(rowInfo),
              })}
              nodeContentRenderer={TreeNodeRenderer}
            />
            <NodeMenu/>
          </Col>
          <Col lg="4" md="4" sm="12">
            <NodeDetails />
          </Col>
        </Row>
      </div>
    );
  }
}

AllocationTree.propTypes = {
  tree: PropTypes.object.isRequired,
};

AllocationTree.defaultProps = {
  tree: {
    name: '',
    data: []
  },
  treeData: [],
};

const mapStateToProps = state => {
  return {
    ...state.allocationTree
  };
};

export default connect( mapStateToProps, {
  getAllocationTree,
  saveAllocationTree,
  selectNode,
  updateTree,
} )(AllocationTree);
