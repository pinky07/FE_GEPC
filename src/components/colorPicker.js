import React from 'react'
import reactCSS from 'reactcss'
import { SketchPicker } from 'react-color'

export default class ColorPicker extends React.Component {

  constructor (props) {
    super(props);
    this.state = {
      displayColorPicker: false,
      color: {
        r: '0',
        g: '32',
        b: '96',
        a: '1',
      },
    };
  }

  handleClick = () => {
    this.setState({ displayColorPicker: !this.state.displayColorPicker })
  };

  handleClose = () => {
    this.setState({ displayColorPicker: false })
  };

  handleChange = (color) => {
    this.setState({ color: color.rgb })
  };

  renderPicker () {
    if (this.state.displayColorPicker) {
      return (
        <div className="popover">
          <div className="cover" onClick={ this.handleClose }/>
          <SketchPicker color={ this.state.color } onChange={ this.handleChange } />
        </div>
      );
    }
    return null;
  }

  render() {

    const styles = reactCSS({
      'default': {
        color: {
          background: `rgba(${ this.state.color.r }, ${ this.state.color.g }, ${ this.state.color.b }, ${ this.state.color.a })`,
        },
      },
    });

    return (
      <div className="colorPicker">
        <div onClick={ this.handleClick }>
          <div className="color" style={ styles.color } />
        </div>
        { this.renderPicker() }
      </div>
    )
  }
}
