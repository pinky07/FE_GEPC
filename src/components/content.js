import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
  Row,
  Col,
  Container,
  TabContent,
  TabPane,
  Nav,
  NavItem,
  NavLink
} from 'reactstrap';
import TreeView from './treeView/treeView';
import GridView from './gridView/gridView';

const TREE_VIEW_TAB = '1';
const GRID_VIEW_TAB = '2';

/**
 * Content component with Grid and Tree View tabs.
 * 
 * @author Francisco ZuÃ±iga
 * @class Content
 * @extends {React.Component}
 */
export class Content extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      activeTab: TREE_VIEW_TAB,
    };
  }

  toggle = index => {
    if (this.state.activeTab !== index) {
      this.setState({
        activeTab: index
      });
    }
  }

  render() {
    return (
      <Container className="content" fluid={true}>
        {this.props.isLoading ? <div className="loader"></div> : null}
        <Row>
          <Col lg="12" md="12">
            <Nav tabs>
              <NavItem>
                <NavLink
                  className={this.state.activeTab === TREE_VIEW_TAB ? 'active' : ''}
                  onClick={() => { this.toggle(TREE_VIEW_TAB); }}
                >
                  Tree View
                </NavLink>
              </NavItem>

              <NavItem>
                <NavLink
                  className={this.state.activeTab === GRID_VIEW_TAB ? 'active' : ''}
                  onClick={() => { this.toggle(GRID_VIEW_TAB); }}
                >
                  Grid View
                </NavLink>
              </NavItem>
            </Nav>
            <TabContent activeTab={this.state.activeTab}>
              <TabPane tabId={TREE_VIEW_TAB}>
                <Row>
                  <Col lg="12">
                    <TreeView />
                  </Col>
                </Row>
              </TabPane>

              <TabPane tabId={GRID_VIEW_TAB}>
                <Row>
                  <Col lg="12">
                    <GridView />
                  </Col>
                </Row>
              </TabPane>
            </TabContent>

          </Col>
        </Row>
      </Container>
    );
  }
}

Content.propTypes = {
  isLoading: PropTypes.bool.isRequired,
};

Content.defaultProps = {
  isLoading: false
};

const mapStateToProps = state => {
  return {
    ...state.shared
  };
};

export default connect(mapStateToProps, null)(Content);
