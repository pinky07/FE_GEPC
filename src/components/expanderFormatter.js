import React from 'react';
import createReactClass from 'create-react-class';
import PropTypes from 'prop-types';

const RIGHT_TRIANGLE = String.fromCharCode('9654');
const DOWN_TRIANGLE = String.fromCharCode('9660');

const ExpanderFormatter = createReactClass({
//&#x25BC;


  getInitialState() {
    let expanded = this.props.expandableOptions && this.props.expandableOptions.expanded;
    return { expanded: expanded };
  },
  componentWillReceiveProps(nextProps) {
    let expanded = nextProps.expandableOptions && nextProps.expandableOptions.expanded;
    if (this.state.expanded !== expanded) {
      this.setState({expanded});
    }
  },

  onCellExpand(e) {
    this.setState({ expanded: !this.state.expanded });
    this.props.onCellExpand(e);
  },
  render() {
    return (
      <span onClick={this.onCellExpand} >
        {this.state.expanded ? DOWN_TRIANGLE : RIGHT_TRIANGLE}
      </span>
    );
  }
});

ExpanderFormatter.propTypes = {
  expandableOptions: PropTypes.object.isRequired,
  onCellExpand: PropTypes.func.isRequired
};

export default ExpanderFormatter;