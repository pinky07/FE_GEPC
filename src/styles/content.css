.nav-tabs {
    padding-top: 1em;
}

.nav-tabs .nav-link {
    color: white !important;
    font-weight: bold;
    background-color: #002060;
}

.currentTab {
    background-color: #a0ac1b !important;
    color: white !important;
}