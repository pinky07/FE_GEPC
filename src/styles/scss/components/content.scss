.nav-tabs {
  padding-top: 1em;
  border-bottom-color: $green;
}

.nav-tabs .nav-link {
  color: white !important;
  font-weight: bold;
  background-color: $blue-1;
}

.currentTab {
  background-color: $green !important;
  color: white !important;
}