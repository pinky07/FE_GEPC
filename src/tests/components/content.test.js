import React from 'react';
import { shallow  } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Content } from '../../components/content';

describe('content component', () => {
  let wrapper = shallow (<Content/>);

  it ('renders correctly', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});