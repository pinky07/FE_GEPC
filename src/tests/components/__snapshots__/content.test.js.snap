// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content component renders correctly 1`] = `
<Container
  className="content"
  fluid={true}
  tag="div"
>
  <Row
    tag="div"
  >
    <Col
      lg="12"
      md="12"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Nav
        tabs={true}
        tag="ul"
      >
        <NavItem
          tag="li"
        >
          <NavLink
            className="active"
            onClick={[Function]}
            tag="a"
          >
            Tree View
          </NavLink>
        </NavItem>
        <NavItem
          tag="li"
        >
          <NavLink
            className=""
            onClick={[Function]}
            tag="a"
          >
            Grid View
          </NavLink>
        </NavItem>
      </Nav>
      <TabContent
        activeTab="1"
        tag="div"
      >
        <TabPane
          tabId="1"
          tag="div"
        >
          <Row
            tag="div"
          >
            <Col
              lg="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Connect(AllocationTree) />
            </Col>
          </Row>
        </TabPane>
        <TabPane
          tabId="2"
          tag="div"
        >
          <Row
            tag="div"
          >
            <Col
              lg="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Connect(Aggrid) />
            </Col>
          </Row>
        </TabPane>
      </TabContent>
    </Col>
  </Row>
</Container>
`;
